cmd/signature-v4.go

```diff
@@ -91,17 +92,18 @@ func getSignedHeaders(signedHeaders http.Header) string {
 //  <HashedPayload>
 //
 func getCanonicalRequest(extractedSignedHeaders http.Header, payload, queryStr, urlPath, method string) string {
-	rawQuery := strings.Replace(queryStr, "+", "%20", -1)
-	encodedPath := getURLEncodedName(urlPath)
-	canonicalRequest := strings.Join([]string{
+	canonicalRequestStrings := []string{
 		method,
-		encodedPath,
-		rawQuery,
+		getURLEncodedName(urlPath),
+		strings.Replace(queryStr, "+", "%20", -1),
 		getCanonicalHeaders(extractedSignedHeaders),
 		getSignedHeaders(extractedSignedHeaders),
-		payload,
-	}, "\n")
-	return canonicalRequest
+	}
+	// If payload is empty do not use it as part of canonical request.
+	if payload != "" {
```

Comment:
Under what situation can the payload be empty string? becausegetContentSha256Cksum() always returns a non-empty string

Reply:
It is not possible .